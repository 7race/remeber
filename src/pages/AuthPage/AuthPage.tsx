import { Button } from '../../atoms/Button/Button';

export const AuthPage = () => (
  <Button variant='success' outlined>
    Войти
  </Button>
);
