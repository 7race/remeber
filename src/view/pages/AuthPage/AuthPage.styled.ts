import styled, { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
  body {
    background: ${({ theme }) => theme.colors.lightGrey};
  }
`;

export const AuthPage = styled.div`
  display: grid;
  height: 100vh;
  grid-template-rows: 20vh minmax(min-content, max-content) 1fr;
  & > form {
    grid-area: form;
  }
  grid-template-areas:
    '.'
    'form'
    '.';
  padding: 20px;

  @media ${({ theme }) => theme.media.tablet} {
    grid-template-rows: 1fr minmax(min-content, max-content) 1fr;
    grid-template-columns: 1fr 400px 1fr;
    grid-template-areas:
      '. . .'
      '. form .'
      '. . .';
  }
`;
