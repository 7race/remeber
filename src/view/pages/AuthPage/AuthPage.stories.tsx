import { ComponentStory, ComponentMeta } from '@storybook/react';
import { AuthPage } from './AuthPage';

export default {
  title: 'AuthPage',
  component: AuthPage,
} as ComponentMeta<typeof AuthPage>;

const Template: ComponentStory<typeof AuthPage> = () => <AuthPage />;

export const Default = Template.bind({});
