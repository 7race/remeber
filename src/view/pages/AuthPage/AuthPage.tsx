import { Form } from '@molecules/Form';
import * as S from './AuthPage.styled';

export const AuthPage = () => (
  <S.AuthPage>
    <S.GlobalStyle />
    <Form />
  </S.AuthPage>
);
