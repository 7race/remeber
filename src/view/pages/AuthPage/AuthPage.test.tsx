import { render } from '@tests-utils';
import { AuthPage } from './AuthPage';

describe('auth page', () => {
  it('correct display', () => {
    const { container } = render(<AuthPage />);
    expect(container).toMatchSnapshot();
  });
});
