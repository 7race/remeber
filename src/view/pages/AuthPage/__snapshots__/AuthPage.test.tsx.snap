// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth page correct display 1`] = `
<div>
  <div
    class="sc-jSMfEi jrJNuz"
  >
    <form
      aria-label="form"
      class="sc-eCYdqJ hPBytf"
    >
      <p
        class="sc-bczRLJ Gujvk"
      >
        Вход
      </p>
      <div>
        <input
          class="sc-dkzDqf dVlRoD"
          placeholder="Логин"
          value=""
        />
        <p
          class="sc-bczRLJ fUulwP text-error"
        >
           
        </p>
      </div>
      <div>
        <div
          class="sc-gsnTZi dGUFFC"
        >
          <input
            class="sc-dkzDqf dVlRoD"
            placeholder="Пароль"
            type="password"
            value=""
          />
          <img
            alt="closed eye"
            src="eye-slash.svg"
          />
        </div>
        <p
          class="sc-bczRLJ fUulwP text-error"
        >
           
        </p>
      </div>
      <div
        class="react-ripples"
        style="position: relative; display: inline-flex; overflow: hidden;"
      >
        <button
          class="sc-hKMtZM blpuEy"
          type="submit"
        >
          Войти
        </button>
        <s
          style="position: absolute; border-radius: 50%; opacity: 0; width: 35px; height: 35px; transform: translate(-50%, -50%); pointer-events: none;"
        />
      </div>
    </form>
  </div>
</div>
`;
